packages/catalog-process/src/services/catalogService.ts

```diff
@@ -197,53 +198,55 @@ export function catalogServiceBuilder(
     async createEService(
       apiEServicesSeed: ApiEServiceSeed,
       authData: AuthData
-    ): Promise<string> {
-      return repository.createEvent(
-        createEserviceLogic({
-          eService: await readModelService.getEServiceByNameAndProducerId({
-            name: apiEServicesSeed.name,
-            producerId: authData.organizationId,
-          }),
-          apiEServicesSeed,
-          authData,
-        })
+    ): Promise<EServiceId> {
+      return unsafeBrandId<EServiceId>(
```

Comment:
I can move the "unsafe cast" inside the `createEvent` function returning an `AgreementId`, an `EServiceId` or other kinds of ID that could be used as `streamId`.

Reply:
if I'm not missing something, the value returned by the `createEvent` function is the streamId inside the parameter passed to the function.
My thought was to use a branded type for this streamId in order to avoid the unsafe cast.
Let me know if this makes sense to you

Reply:
It's not easy to extract the ID brand type from the event payload. I have an idea but I prefer to create a dedicated PR.